x/httpproxy/proxy_handler.go

```diff
@@ -22,6 +22,7 @@ import (
 
 type proxyHandler struct {
 	connectHandler http.Handler
+	pathHandler    http.Handler
 	forwardHandler http.Handler
 }
```

Comment:
I'd like to change our approach here. The proxyHandler is currently a standard web proxy. The path extension is not. I'd like to keep having a standard proxy, and once we add something, we can't remove it. I'm also not a fan of hardcoding a path for the path handler.

Let's have a fallback handler instead:
```suggestion
type ProxyHandler struct {
	connectHandler http.Handler
	forwardHandler http.Handler
	fallbackHandler http.Handler
}
```

Make `NewProxyHandler` return `ProxyHandler`, and add a `SetFallbackHandler(handler)` method, which should be the [`http.NotFoundHandler()`](https://pkg.go.dev/net/http#NotFoundHandler) by default.

Then, in the Mobileproxy code, add the pathHandler here:
https://github.com/Jigsaw-Code/outline-sdk/blob/eec005b32f3c01a0b3beaac18edcd56e489a3762/x/mobileproxy/mobileproxy.go#L80

Let's put the path handler on a subpath, so we can add other handlers in the future (for example, a health check or status endpoint, perhaps an API).

You can use [http.ServeMux](https://pkg.go.dev/net/http#ServeMux) to register a handler on a path, combined with [http.StripPrefix](https://pkg.go.dev/net/http#StripPrefix) to remove the path prefix from the request.

Perhaps we go with `/proxy/` for now? Ideally it would be configurable, but we can figure that our later.

Reply:
Actually, for this PR there's no need to include the mobileproxy. We can add that in another PR.
For this PR:
- keep the path_handler (with the fixes)
- revert the proxy_handler
- Add the fallback support either in this PR or another PR